tests/src/JIT/jit64/mcc/interop/native_i1c.cpp

```diff
@@ -4,25 +4,20 @@
 #include <stdarg.h>
 #include "native.h"
 
-#ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wvarargs"
-#endif
-
-MCC_API VType1 sum(float first, ...) {
+MCC_API VType1 sum(double first, ...) {
```

Comment:
This is not the same test anymore. Was the previous test not valid or unnecessary?

Reply:
Three compilers have different issues with the implementation.

For Clang & GCC, see: https://stackoverflow.com/a/24967341

clang had no issues with `va_list`, but warned on `va_start`: `warning: passing an object that undergoes default argument promotion to 'va_start' has undefined behavior`

gcc had no issue with `va_start`, but warned about promotion in `va_list` (implied that `va_list`'s second argument should explicitly be `double`).

MSVC required more casts: `float x = (float)doubleValue` (note original version already have few of them).

> This is not the same test anymore.

If according to C99 and C11 standards (which C++ include by reference), these promotions will take place under the wraps, then this change only makes the compilers' implicit behavior explicit, no?

Reply:
>Was the previous test not valid or unnecessary?

Seems that the previous test wasn't really valid as written. I didn't know the promotion rules regarding varargs - thanks. This change makes sense then as I agree making the argument type explicit is a laudable goal. Someone from @dotnet/jit-contrib should review this test change as well.